internal/devicefinder/discovery/discovery.go

```diff
@@ -143,30 +143,29 @@ func (discovery *DeviceDiscovery) discoverDevices() error {
 	return nil
 }
 
-// getValidBlockDevices fetchs all the block devices sutitable for discovery
-func getValidBlockDevices() ([]diskutil.BlockDevice, error) {
-	blockDevices, output, err := diskutil.ListBlockDevices([]string{})
+// getValidBlockDevices fetches and unmarshalls all the block devices sutitable for discovery
+func getValidBlockDevices() ([]diskutils.BlockDevice, error) {
+	lDevices := diskutils.BlockDeviceList{}
+	blockDevices, err := diskutils.GetBlockDevices()
 	if err != nil {
-		return blockDevices, errors.Wrapf(err, "failed to list all the block devices in the node, stderr=%v", output)
+		return lDevices.BlockDevices, err
```

Comment:
Better to return `nil,err`. There is nothing to gain from returning whatever is being returned by `diskutils.GetBlockDevices()` when an error has been triggered. The go idiomatic way is just to return a nil or defalut value("",0,false, etc...) and the error just to avoid the potential scenario that the error is discarded by the caller whilst the other parameter is used and contains invalid data.